mslib/msui/wms_control.py

```diff
@@ -556,7 +577,13 @@ def initialise_wms(self, base_url):
             str(base_url)  # to provoke early Unicode Error
             while wms is None:
                 try:
-                    wms = MSSWebMapService(base_url, version='1.1.1',
+                    # Version could not be detected without authentication
+                    if not version:
+                        response = openURL(base_url, "service=WMS&request=GetCapabilities",
```

Comment:
Does this generate a third request to the server? Cannot we fold that into the class instantiation below? We will use the version the server offers, won't we?

Reply:
I changed it so the class handles dynamic versions, not wms_control. This gets rid of the third request.